devsetup/tripleo/ceph.sh

```diff
@@ -38,14 +38,6 @@ data_devices:
     - /dev/ceph_vg0/ceph_lv0
 EOF
 
-# create roles file
-openstack overcloud roles generate Controller ComputeHCI > roles.yaml
-
-# disable external gateway for controller nodes
-sed -i "s/default_route_networks: \['External'\]/default_route_networks: \['ControlPlane'\]/" roles.yaml
-sed -i "/External:/d" roles.yaml
-sed -i "/subnet: external_subnet/d" roles.yaml
-
 # NOTE: TripleO has the hardcoded --yes-i-know option that is not valid anymore
```

Comment:
@marios @cescgina 

The status quo is:
* tripleo netconfig It IS no longer skipped (but I have to do that for my local setup to work!)
* however, we should NOT configure default route overrides for CI, or it fails reaching out nodes via FIPs...
* Ceph deploy script still does remove the external network attachement for the controller role in tripleo roles file, other multinode non-ceph jobs keep it in place

I wonder now, how should I align things now, to let my local setup and CI work happily?..
Maybe I will need an extra var like TRIPLEO_NETCONFIG_CONFIGURE_EXTNET_ROUTE and set it to false for CI only ?

Reply:
@bogdando I'm a bit OOTL on this one, do you have a run where the problem is happening? Looking at the patch, https://github.com/openstack-k8s-operators/install_yamls/pull/928/files#diff-6d137e1f40b93f9957103127166db87dbd340c674dcf1d8694def950ebd56821R169 removes the external network if `TRIPLEO_NETWORKING` is true, which it is by default https://github.com/openstack-k8s-operators/install_yamls/pull/928/files#diff-66217dcd52416a1c2c137246eae386a8a42ac651b6c7e30c5937757764675b1bR100

Reply:
I set it to true by default to meet the needs of local setups. In multinode adoption CI it used to be FALSE, befor recently (as I occationally figured "the switch" out)

Reply:
in CI there is a pre-configured netconfig https://github.com/bogdando/ci-framework/blob/5b3188412eb476b1b3831f3e4168f00a488d6f6c/roles/adoption_osp_deploy/tasks/prepare_overcloud.yml#L236, this is why originally, multinode CI wasn't using the tripleo netconfig driven by ansible.
I tried to make it working in the "overwrite" mode, but it never worked for my multi-cell testing. Anyway, now I see that build logs DO apply the tripleo network config tasks in multi node CI, not sure where from that change landed, @marios do you know?

Reply:
hm, I am still seeing the netconfig for tripleo disabled in CI https://github.com/bogdando/install_yamls/blob/263f88f5f2b6d0c80be59f9f02ebe0795e8ec389/devsetup/tripleo/overcloud_services.yaml.j2#L54, I have to wrap my head around all that...